### PR TITLE
docs: extend using proxy section

### LINKED ISSUE #3706: [DOCS] Extend `Using a proxy` section in docs
Extend the [Using a proxy](https://docs.argilla.io/en/latest/getting_started/installation/configurations/server_configuration.html#using-a-proxy) section with examples on how to setup an NGINX and Traefik proxy to forward requests to the Argilla server.

### PR DESCRIPTION
**Pull request by gabrielmbmb**
# Description

This PR adds examples to the docs on how to use `NGINX` and `Traefik` as reverse proxy for Argilla server.

Closes #3706

**Type of change**

- [x] Documentation update

**How Has This Been Tested**

- [x] `sphinx-autobuild` (read [Developer Documentation](https://docs.argilla.io/en/latest/community/developer_docs.html#building-the-documentation) for more details)

**Checklist**

- [x] I added relevant documentation
- [x] I followed the style guidelines of this project
- [x] I did a self-review of my code
- [x] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: docker/nginx/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  nginx:
+    image: nginx:latest
+    container_name: "nginx_proxy"
+    ports:
+      - "80:80"
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf:ro
+
+  argilla:
+    image: argilla/argilla-quickstart:v1.20.0
```
Comment (davidberenstein1957) [inline docker/nginx/docker-compose.yaml]: this should be latest right?

File: docker/nginx/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    environment:
+      ARGILLA_ENABLE_TELEMETRY: 0
```
Comment (davidberenstein1957) [inline docker/nginx/docker-compose.yaml:15]: We don't want to disable this by default I think

File: docker/traefik/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      - "--providers.docker.exposedbydefault=false"
+      - "--entrypoints.web.address=:80"
+    ports:
+      - "80:80"
+      - "8080:8080"
+    volumes:
+      - "/var/run/docker.sock:/var/run/docker.sock:ro"
+
+  argilla:
+    image: argilla/argilla-quickstart:v1.20.0
```
Comment (davidberenstein1957) [inline docker/traefik/docker-compose.yaml]: Same here

File: docker/traefik/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    environment:
+      ARGILLA_ENABLE_TELEMETRY: 0
```
Comment (davidberenstein1957) [inline docker/traefik/docker-compose.yaml:22]: same here

**Review by davidberenstein1957 — COMMENTED** [review]
Hi @gabrielmbmb , some small remarkts

File: docker/nginx/docker-compose.yaml (hunk 1 above)
Comment (gabrielmbmb) [inline docker/nginx/docker-compose.yaml]:
```suggestion
    image: argilla/argilla-quickstart:latest
```

File: docker/traefik/docker-compose.yaml (hunk 3 above)
Comment (gabrielmbmb) [inline docker/traefik/docker-compose.yaml]:
```suggestion
    image: argilla/argilla-quickstart:latest
```

File: docker/nginx/docker-compose.yaml (hunk 2 above)
Comment (gabrielmbmb) [reply docker/nginx/docker-compose.yaml:15]: This is supposed to be an example, not sure if we want to enable telemetry here, right?

**Pull request merged**